lib/trip_dispatcher.rb

```diff
@@ -67,27 +68,82 @@ def load_trips
       trips = []
       trip_data = CSV.open('support/trips.csv', 'r', headers: true, header_converters: :symbol)
 
+      # header_converters: :symbol means to take the first line of headers
+      # to make into keys
+      # taking the raw data from the CSV
       trip_data.each do |raw_trip|
         driver = find_driver(raw_trip[:driver_id].to_i)
         passenger = find_passenger(raw_trip[:passenger_id].to_i)
 
+        if driver.nil? || passenger.nil?
+          raise "Could not find driver or passenger trip ID #{raw_trip[:id]}"
+        end
+
+        # setting up parsed_trip to take in the raw_trip data
+        # prepping it for the Trip class
         parsed_trip = {
           id: raw_trip[:id].to_i,
           driver: driver,
           passenger: passenger,
-          start_time: raw_trip[:start_time],
-          end_time: raw_trip[:end_time],
+          start_time: Time.parse(raw_trip[:start_time]),
+          end_time: Time.parse(raw_trip[:end_time]),
           cost: raw_trip[:cost].to_f,
           rating: raw_trip[:rating].to_i
         }
 
+        # this setting up the instance of trip
         trip = Trip.new(parsed_trip)
+
+        # Set up relations
         driver.add_trip(trip)
         passenger.add_trip(trip)
         trips << trip
       end
 
-      trips
+      return trips
+    end
+
+    # this method goes through and finds the first driver with status that is :AVAILABLE
+    def available_driver
+      @drivers.find do |driver|
+        if driver.status == :AVAILABLE
+          return driver
+        end
+      end
+      raise ArgumentError.new("There are no drivers available at this time")
+    end
+
+    def request_trip(passenger_id)
+
+      passenger = find_passenger(passenger_id)
+
+      if passenger.nil?
+        raise "Could not find passenger trip ID #{passenger_id}"
+      end
```

Comment:
This error message talks about a trip ID, but interpolates the passenger ID. Which is it?